### DISCUSSION TITLE
How to generate OpenGraph/Twitter Card Image for every article

### DISCUSSION THREAD

**Discussion by jcubic** [11ty/eleventy#2995] [Q&A] [answered]
I need to add image for every post, limited by tags I use. I've created an SVG file that I can use as template, where I will add
`{{ title }}` and `{{ users[author].name }}`. I only don't know how to generate this image for every md file I have.

[![](https://raw.githubusercontent.com/jcubic/jankiewicz/0efc40486bf1365adda129bf8fe38ab6c4420858/static/img/card.svg)](https://github.com/jcubic/jankiewicz/blob/master/static/img/card.svg)

this is my image, that I want to render as png or jpg using [rsvg](https://en.wikipedia.org/wiki/Librsvg) tool when building the website.

I was thinking about using transforms for that, but I will need to open the source file and parse it myself to extract the title and tags.

I was looking for [Zach](https://www.zachleat.com/) blog but it use some kind of service for this, he don't generate images with Eleventy.

**Comment by jcubic** [comment, marked as answer]
So if anyone is interested, now or in the future. I've created shorcode for this. And to render the image that include foreignObject and p tag, so the title wraps into multiple lines I used Pappetier.

This is the commit that add this https://github.com/jcubic/jankiewicz/commit/c1642a4396390ad9836ce3c38c5abb6c14735fc1

**Comment by pdehaan** [comment]
> I was looking for [Zach](https://www.zachleat.com/) blog but it use some kind of service for this, he don't generate images with Eleventy.

I think there is https://www.11ty.dev/docs/services/screenshots/

**Reply by jcubic** [reply]
I don't want to use such service. I prefer to have control over my own images.
